src/app/components/Application/Application.jsx

```diff
@@ -37,14 +48,23 @@ class Application extends React.Component {
 
   componentDidMount() {
     Store.listen(this.onChange);
-
-
     window.addEventListener('resize', this.onWindowResize.bind(this));
     this.onWindowResize();
-  }
-
-  shouldStoreUpdate() {
-    return `?${basicQuery({})(Store.getState())}` !== this.context.router.location.search;
+    const { router } = this.context;
+    if (this.state.urlEnabledFeatures) {
+      router.listen(() => {
+        const {
+          pathname,
+          query,
+        } = router.location;
+        if (query.features !== this.state.urlEnabledFeatures) {
+          router.replace({
+            pathname,
+            query: Object.assign(query, { features: this.state.urlEnabledFeatures }),
+          });
```

Comment:
Yes. The listener intercepts the route change and replaces it with the same route with the `feature` param included in the query/search, before the `urlEnabledFeatures` is set to no feature. This is just something I came up with EOD yesterday. So, a little proud that it actually works, but also acknowledging it is not vouched for...

Reply:
Granted this is clever but it seems inefficient. If I understand correctly, this means every link will now involve two page loads instead of one?

Reply:
It doesn't look to me like it's actually creating two page loads in practice. The hook intercepts the navigation before data is fetched and before anything begins to render. So I'm not sure efficiency is an issue. I was concerned about how this would behave with POST'd pages (because location.replace issues a GET?) but this method seems to persist the feature flag in the URL across the HoldRequest workflow.

The only downside I see to this method over a cookie at this point is it may not be obvious to a tester that they have to remain withing SCC to retain the feature (they can't navigate to other areas via the header). But that may be a better problem to have than the different problems created by a cookie (like making it simple to remove).

Reply:
I checked how many times `Application` renders for a couple page transitions and on this branch, it remains consistent from the `development` branch.